plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -412,3 +429,14 @@ func MapPodsToMachines(lister algorithm.PodLister) (map[string][]*api.Pod, error
 	}
 	return machineToPods, nil
 }
+
+func haveSame(a1, a2 []string) bool {
```

Comment:
Is this func supposed to test the equality (contain the same elements) of the string arrays?

This implementation will return true on the first matching pair of values and ignore the rest of the elements of the array.